samples/gen-js/index.js

```diff
@@ -0,0 +1,391 @@
+
+const discovery = require("@clever/discovery");
+const request = require("request");
+const url = require("url");
+const opentracing = require("opentracing");
+
+// go-swagger treats handles/expects arrays in the query string to be a string of comma joined values
+// so...do that thing. It's worth noting that this has lots of issues ("what if my values have commas in them?")
+// but that's an issue with go-swagger
+function serializeQueryString(data) {
+  if (Array.isArray(data)) {
+    return data.join(",");
+  }
+  return data;
+}
+
+module.exports = class SwaggerTest {
+
+  address;
+  timeout;
+
+  constructor(options) {
+    options = options || {};
+
+    if (options.discovery) {
+      this.address = discovery("swagger-test", "default").url();
+    } else if (options.address) {
+      this.address = options.address;
+    } else {
+      throw new Error("Cannot initialize swagger-test without discovery or address");
+    }
+    if (options.timeout) {
+      this.timeout = options.timeout
+    }
+  }
+
+  
+  getBooks = (params, options, cb) => {
+
+    if (!cb && typeof options === "function") {
+      cb = options;
+      options = undefined;
+    }
+
+    if (!options) {
+      options = {};
+    }
+
+    const timeout = options.timeout || this.timeout;
+    const span = options.span;
+
+    const headers = {
+
+    }
+
+    if (span) {
+      opentracing.inject(span, opentracing.FORMAT_TEXT_MAP, headers);
+      span.logEvent("GET /v1/books");
+    }
+
+    const requestOptions = {
+      method: "GET",
+      uri: this.address + "/v1/books",
+      json: true,
+      timeout,
+      headers,
+      body: {
+
+      },
+      qs: {
+
+        "authors": serializeQueryString(params.authors),
+
+        "available": serializeQueryString(params.available),
+
+        "state": serializeQueryString(params.state),
+
+        "published": serializeQueryString(params.published),
+
+        "snake_case": serializeQueryString(params.snakeCase),
+
+        "completed": serializeQueryString(params.completed),
+
+        "maxPages": serializeQueryString(params.maxPages),
+
+        "min_pages": serializeQueryString(params.minPages),
+
+        "pagesToTime": serializeQueryString(params.pagesToTime),
+
+      },
+    };
+    return new Promise((resolve, reject) => {
+      const rejecter = (err) => {
+        reject(err);
+        if (cb) {
+          cb(err);
+        }
+      }
+      const resolver = (data) => {
+        resolve(data);
+        if (cb) {
+          cb(null, data);
+        }
+      }
+
+      request(requestOptions, (err, response, body) => {
+        if (err) {
+          return rejecter(err);
+        }
+        if (response.statusCode >= 400) {
+          return rejecter(new Error(body));
+        }
+        resolver(body);
+      });
+    });
+  }
+
+  
+  getBookByID = (params, options, cb) => {
```

Comment:
this doesn't quite follow what I laid out in slack--since this method only has one input (`bookID`), the signature should be `(bookID, options, cb)`